compiler/src/typed/translprim.re

```diff
@@ -34,6 +40,12 @@ let pat_c = mkpatvar("c");
 let prim_map =
   PrimMap.of_seq(
     List.to_seq([
+      ("@heap.base", PrimitiveWasmI32(HeapBase)),
+      ("@heap.start", PrimitiveWasmI32(HeapStart)),
+      (
+        "@heap.type_metadata_pointer",
```

Comment:
I think we can drop the `_pointer` part to match the other two, with `HeapTypeMetadata`.